### ISSUE TITLE
Theme for admin

### ISSUE THREAD

**Issue by rosa2** [iomad/iomad#325]
Hello

I am learning theme in Iomad and the admin user always gets the iomad theme, even if I assign its subtheme, that I created, to the site. No idea where to look. Any tips :) ?

**Comment by rosa2** [comment]
I put here all my doubts, so I don't fill the issue cue :)

I found themes/iomad/style/custom.css and themes/iomad/style/company.css. They don't have themes/iomad/less/custom.less and themes/iomad/less/company.less. Is that because they are written from somewhere else? Any reason? Thanks

If I compile with grunt, I get this error:

> > SyntaxError: unexpected token in theme/intermedia/less/custom.less on line 21, column 2:
> > 20 -------------------------*/
> > 21 [[setting:customcss]]
> > 22
> > Warning: Error compiling theme/intermedia/less/custom.less Use --force to continue.

Aborted due to warnings.

So I imagine that those have to be css and not less :)

**Comment by turf212** [comment]
You may either need to refresh the theme cache or may need to log out and back in again to pick up the theme change.  Failing that, if you have added the user to a company then they will pick up the company theme (which defaults to Iomad).

The styles were put in by hand and not added to the Less files.  I guess they really should be in there....

**Comment by rosa2** [comment]
I have cleared all caches (site and theme). When I am not logged in, I see the subtheme that I created, but when I log in as admin, I always see the Iomad theme. If I go to a company, then I see the subtheme well. That is a solution, but it would be nice to be able to see the general subtheme, especially because I am changing the theme and also, because is... confusing for new users :)

**Comment by rosa2** [comment]
I have in my subtheme general the logo for the general organization and each company has its own logo choosen when admin was in the co. Besides that I change the background color to easily see if the subtheme is taken. 
I have done test with the roles that I need:
- anonymous: sees subtheme and general logo on the left
- admin and general manager: when they log in, they see iomad theme. When they go to a company, they see the iomad theme and the company logo on the right.
- company manager, teacher and student: they see subtheme, general logo on the left and they don't see the company logo.

So I think the patch I made makes possible to see the companies logos only on Iomad theme. Any tips to fix this, please? Thanks

SOLVED: I went to admin profile and could change the theme :) Sorry.

**Comment by rosa2** [comment]
I think that the patch I gave only shows the form in the subtheme, but the content gets save in Iomad theme. I found a solution for my subtheme: in lib.php, line130, I don't substitute theme_iomad by my theme_subthemeName and it works :)

if ($companyid = iomad::is_company_user()) {
        $context = context_system::instance();

```
        if ($files = $DB->get_records('files', array('contextid' => $context->id, 'component' => 'theme_iomad', 'filearea' => 'companylogo', 'itemid' => $companyid))) {
            foreach ($files as $file) {
                if ($file->filename != '.') {
                    $clientlogo = $CFG->wwwroot . "/pluginfile.php/{$context->id}/theme_iomad/companylogo/$companyid/{$file->filename}";
                }
            }
        }
```

**Comment by rosa2** [comment]
Did you create the iomad.css from the iomad.less?  I don't understand that in iomad.less there is not and in iomad.css there is:

```
#sitelogo {
    float: left;
    width:100px;
    padding-bottom: 10px;
}

#siteheading {
    float: left;
    left: 100px;
}

#clientlogo {
    float: right;
    width: 100px;
}

/* This hides Authentication Method from edit user form which is redundant when using IOMAD and potentially causes issues */
#fitem_id_auth{
    display: none;
}
```

Also the font is not compiled

```
@font-face {
  font-family: iomad;
  src: url([[font:theme|astonish.woff]]);
}
```

If you need that I pass the styles to iomad.less and compile with less and give to you the files, I will be pleased :)

Is this function to resolve this? In lib.php, line 35

```
function theme_iomad_process_css($css, $theme) {
    global $CFG;

    // Set custom CSS.
    if (!empty($theme->settings->customcss)) {
        $customcss = $theme->settings->customcss;
    } else {
        $customcss = null;
    }
    $css = theme_iomad_set_customcss($css, $customcss);

    // deal with webfonts
    $tag = '[[font:theme|astonish.woff]]';
    $replacement = $CFG->wwwroot.'/theme/iomad/fonts/astonish.woff';
    $css = str_replace($tag, $replacement, $css);

    return $css;
}
```

Have a good night/day :)

**Comment by rosa2** [comment]
I don't find the way for admin and manager to see the subtheme. They always get the Iomad theme. Any tips are wellcome :)

**Comment by rosa2** [comment]
It was my mistake that the theme for admin was Iomad: I went to user admin profile and saw that Iomad was selected. Sorry for the confusion :)

**Comment by turf212** [comment]
The Iomad CSS were added in directly.  If you have the required additions for the less files that would be fabulous and I'll add them to core.

**Comment by rosa2** [comment]
I have installed the required additions for less files and I will be pleased to return to you some of much I take with Iomad that you developed. But before doing, I would like to know why do you have two themes. In my opinion, I will have only Iomad theme that it is much more complete and lets company modifications (css and logo) and will help to maintain. Also, in Iomad theme, I will remove some css that are not used (all that is in custom.css and in company.css since they use the logo as a background and never are aplied). 

To add appearance settings that are selected in the edit company form you only need in theme config.php 

`$THEME->extralesscallback = 'theme_iomad_extra_less';`

and add the function in theme lib.php

```
function theme_iomad_extra_less($theme) {
    return $theme->settings->customless;
}
```

For people that doesn't want to use less, I will keep in theme config.php (I have tested including in a company background color change in edit company form and it works)
`
$THEME->csspostprocess = 'theme_iomad_process_css';`

Also, I will add all the iomad.css to iomad.less and modify the 

```
#clientlogo {
    float: right;
    width: 100px;
}
```

to

```
#clientlogo {
    float: right;
    width: auto;
}
```

because it makes the logo to be out of the header on the right.

At last, I will create for .less 2 separate files for variables and typography that will be included in the iomad.less and will facilitate people to have easy editing.

Please, tell to me what you thing and I will make :)
As you know better iomad, the only thing that it is left is that in the edit company, instead to save to iomad theme the settings, they would be saved in the subtheme :)

Thanks for Iomad :)

**Comment by rosa2** [comment]
I made a new issue https://github.com/iomad/iomad/issues/327 for the subtheme saving the logos no to iomad. Here we can follow the less topic :)

**Comment by rosa2** [comment]
I had to work in other subject. I retake this now :)

I upload 2 images to facilitate comunication about the clientlogo position.

```
#clientlogo {
    float: right;
    width: 100px;
}
```

to

```
#clientlogo {
    float: right;
    width: auto;
}
```

![iomad_logo_default](https://cloud.githubusercontent.com/assets/1841507/14674010/b9d86036-0701-11e6-98c7-f3e147e25067.png)
![iomad_logo_corrected](https://cloud.githubusercontent.com/assets/1841507/14674011/b9df75c4-0701-11e6-869e-6a1c187a618f.png)

**Comment by turf212** [comment]
The reason why there are two themes is one was developed directly for Iomad and the other was given to us to allow for a more corporate look and feel at set up.  If you have the less additions I'd love to have them.  It's give more options for those coming later. :)

**Comment by rosa2** [comment]
Now that I have learned some about theming in Moodle, I would like to use this theme to be able to take more ability using the 3.x version (I have done only two sites with it).

I just went to test again and there are 2 thing to be fixed for it to work:
- Custom CSS field in settings
- Custom company CSS in the company settings (logo works).

Because of them I said that it didn't work :) and didn't used it anymore. Now I understand better. Since one is the same on Iomad, I can fix both. 

For not to be confusing, I propose to use one thread where it is the general follow and then others with points once you tell to me ok :)

Very good the theming system in Moodle. I hope to get jobs to be able to learn it :)
